From 30ec9d2b0cd9d5b44fc87126c3df43485c599460 Mon Sep 17 00:00:00 2001
From: George G <kreach3r@users.noreply.github.com>
Date: Tue, 3 Nov 2015 23:57:22 +0200
Subject: [PATCH 4/4] Settings: Expose storage summary text

forward ported and adapted to oreo

This commit simply exposes the storage summary text hardcodes to allow
themers to theme them without issue.

nathanchance edit: updated for Nougat 7.1

Change-Id: I10c124b6a1516a1be7ac234c1dd7dbe2cbe05bb2
Signed-off-by: Adarsh-MR <adarshmr1998@gmail.com>

Conflicts:
	res/layout/storage_summary.xml
---
 res/layout/storage_summary.xml                                    | 6 +++---
 res/values/custom_colors.xml                                      | 4 ++++
 src/com/android/settings/deviceinfo/StorageSummaryPreference.java | 7 +++++--
 src/com/android/settings/deviceinfo/StorageVolumePreference.java  | 5 ++++-
 4 files changed, 16 insertions(+), 6 deletions(-)

diff --git a/res/layout/storage_summary.xml b/res/layout/storage_summary.xml
index 1beb49f..fa6df6c 100644
--- a/res/layout/storage_summary.xml
+++ b/res/layout/storage_summary.xml
@@ -33,10 +33,10 @@
         android:singleLine="true"
         android:textAlignment="viewStart"
         android:textAppearance="@android:style/TextAppearance.Material.Subhead"
-        android:textColor="?android:attr/colorAccent"
-        android:textSize="36sp"
         android:ellipsize="marquee"
-        android:fadingEdge="horizontal" />
+        android:fadingEdge="horizontal"
+        android:textColor="@color/storage_summary_title_color"
+        android:textSize="36sp"/>
 
     <TextView
         android:id="@android:id/summary"
diff --git a/res/values/custom_colors.xml b/res/values/custom_colors.xml
index 8b73612..d6895c4 100644
--- a/res/values/custom_colors.xml
+++ b/res/values/custom_colors.xml
@@ -24,4 +24,8 @@
     <color name="storage_private_3">#fff2a600</color>
     <color name="storage_private_4">#ffec407a</color>
     <color name="storage_private_5">#ffc0ca33</color>
+
+    <!-- Storage Summary Title -->
+    <color name="storage_summary_title_color">@*android:color/accent_device_default_light</color>
+
 </resources>
diff --git a/src/com/android/settings/deviceinfo/StorageSummaryPreference.java b/src/com/android/settings/deviceinfo/StorageSummaryPreference.java
index f97f049..4cb47a1 100644
--- a/src/com/android/settings/deviceinfo/StorageSummaryPreference.java
+++ b/src/com/android/settings/deviceinfo/StorageSummaryPreference.java
@@ -24,17 +24,20 @@ import android.util.MathUtils;
 import android.view.View;
 import android.widget.ProgressBar;
 import android.widget.TextView;
-
+import com.android.settings.Utils;
 import com.android.settings.R;
 
 public class StorageSummaryPreference extends Preference {
     private int mPercent = -1;
+    private int mSecondaryColor;
 
     public StorageSummaryPreference(Context context) {
         super(context);
 
         setLayoutResource(R.layout.storage_summary);
         setEnabled(false);
+
+        mSecondaryColor = Utils.getColorAttr(context, android.R.attr.textColorSecondary);
     }
 
     public void setPercent(long usedBytes, long totalBytes) {
@@ -54,7 +57,7 @@ public class StorageSummaryPreference extends Preference {
         }
 
         final TextView summary = (TextView) view.findViewById(android.R.id.summary);
-        summary.setTextColor(Color.parseColor("#8a000000"));
+        summary.setTextColor(mSecondaryColor);
 
         super.onBindViewHolder(view);
     }
diff --git a/src/com/android/settings/deviceinfo/StorageVolumePreference.java b/src/com/android/settings/deviceinfo/StorageVolumePreference.java
index f8f6fb1..b6e3fc9 100644
--- a/src/com/android/settings/deviceinfo/StorageVolumePreference.java
+++ b/src/com/android/settings/deviceinfo/StorageVolumePreference.java
@@ -46,6 +46,7 @@ public class StorageVolumePreference extends Preference {
     private final VolumeInfo mVolume;
 
     private int mColor;
+    private int mSecondaryColor;
     private int mUsedPercent = -1;
 
     // TODO: ideally, VolumeInfo should have a total physical size.
@@ -56,6 +57,8 @@ public class StorageVolumePreference extends Preference {
         mVolume = volume;
         mColor = color;
 
+        mSecondaryColor = Utils.getColorAttr(context, android.R.attr.textColorSecondary);
+
         setLayoutResource(R.layout.storage_volume);
 
         setKey(volume.getId());
@@ -109,7 +112,7 @@ public class StorageVolumePreference extends Preference {
     public void onBindViewHolder(PreferenceViewHolder view) {
         final ImageView unmount = (ImageView) view.findViewById(R.id.unmount);
         if (unmount != null) {
-            unmount.setImageTintList(ColorStateList.valueOf(Color.parseColor("#8a000000")));
+            unmount.setImageTintList(ColorStateList.valueOf(mSecondaryColor));
             unmount.setOnClickListener(mUnmountListener);
         }
 
-- 
2.1.2

